assets/prometheus-user-workload/prometheus.yaml

```diff
@@ -211,6 +231,8 @@ spec:
     fsGroup: 65534
     runAsNonRoot: true
     runAsUser: 65534
+    seccompProfile:
```

Comment:
@simonpasquier @jan--f So after a lot of debugging I've concluded that this is the problematic part. If we set a `seccompProfile` then we cannot set `runAsUser` with `65534` since "`must be in the ranges: [1000440000, 1000449999]`". I see this error in the event of the UWM namespace with an error for a failed creation of a pod. 

I thought about also removing the `runAsUser` but from what I understood in [this comment](https://github.com/openshift/cluster-monitoring-operator/blob/master/jsonnet/components/prometheus-user-workload.libsonnet#L139-L146), it's necessary and cannot be removed.

Sooo now I'm thinking we should either set the Pod Security Admission labels to `baseline` on the namespace (`baseline` does not enforce `seccompProfile`). Or use a User in the range??

Reply:
Afaiu there nothing technical special about the `65534` uid, its just a convention that typically means `nobody`. So using a uid in the required range would be the ideal solution imho.
I might be missing some history here though, so curious to see what @simonpasquier can add here.

Reply:
I'd be wary to change this setting. The full history behind it is in https://bugzilla.redhat.com/show_bug.cgi?id=1868976 and https://github.com/openshift/cluster-monitoring-operator/pull/981 but the TL;DR is that to be immune to cluster admins adding new SCCs, workloads running with persistent storage need to be associated with the `nonroot` SCC so they get assigned predictable uid and fsGroup values. I would assume that `baseline` is acceptable since monitoring is a core OCP component.

Reply:
After some more debugging I found out that everything works correctly if we use `nonroot-v2` instead of `nonroot`. I'm not quite sure why k8s is not taking into account `nonroot` however...

Reply:
`nonroot-v2` is something new to me. Can you shed some light on how it differs from `nonroot`? 

> Now my question is, is it supported by OCP that a user changes the restricted SCC? Couldn't the user use instead the nonroot SCC?

IIRC though it's strongly advised not to modify the default SCCs [1] but the issue was also happening when users created new SCCs (check [comment 34](https://bugzilla.redhat.com/show_bug.cgi?id=1868976#c34)).

[1] https://docs.openshift.com/container-platform/4.10/authentication/managing-security-context-constraints.html#default-sccs_configuring-internal-oauth